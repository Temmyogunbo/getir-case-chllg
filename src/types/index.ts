export enum CheckBoxType {
  Radio = 'Radio',
  Square = 'Square',
}

export enum ProductType {
  mug = 'mug',
  shirt = 'shirt',
}

export interface IProduct {
  price: string;
  name: string;
  description: string;
  manufacturer: string;
  itemType: ProductType;
  tags: string[];
  added: number;
  slug: string;
}

export interface ICompany {
  slug: string;
  name: string;
  address: string;
  city: string;
  state: string;
  zip: string;
  account: number;
  contact: string;
}

export enum SortBy {
  PRICE_LOW_TO_HIGH = 'PRICE_LOW_TO_HIGH',
  PRICE_HIGH_TO_LOW = 'PRICE_HIGH_TO_LOW',
  NEW_TO_OLD = 'NEW_TO_OLD',
  OLD_TO_NEW = 'OLD_TO_NEW',
}

export interface IOrder extends IProduct {
  count?: number;
}

export interface IInitialState {
  orderReducer: null | any;
  tagReducer: null | any;
  productReducer: null | any;
  companyReducer: null | any;
  brandReducer: null | any;
  sortReducer: null | any;
  pageReducer: null | any;
  itemTypeReducer: null | any;
}

export enum Tags {
  All = 'All',
  Trees = 'Trees',
  Beach = 'Beach',
  Ocean = 'Ocean',
  Water = 'Water',
  Animal = 'Animal',
  Bear = 'Bear',
  Road = 'Road',
  Rocks = 'Rocks',
  Sunset = 'Sunset',
  Rust = 'Rust',
  Old = 'Old',
  Car = 'Car',
  Coffee = 'Coffee',
  Fog = 'Fog',
  People = 'People',
  Dock = 'Dock',
  Person = 'Person',
  Hills = 'Hills',
  Animals = 'Animals',
  Sheep = 'Sheep',
  Lake = 'Lake',
  Building = 'Building',
  Metal = 'Metal',
  Woman = 'Woman',
}

export enum ItemTypes {
  Shirt = 'shirt',
  Mug = 'mug',
}
