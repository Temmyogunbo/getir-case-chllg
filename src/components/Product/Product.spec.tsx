
import { Product } from './index';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ItemTypes } from '../../types';

describe('Procduct Component', () => {
  const product = {
    price: '10',
    name: 'test',
    description: 'test',
    manufacturer: 'test',
    itemType: ItemTypes.Mug,
    tags: 'Trees',
    added: 12345,
    slug: 'test',
  }
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Product product={product} />,
    );

    expect(asFragment()).toMatchSnapshot();
  })
})