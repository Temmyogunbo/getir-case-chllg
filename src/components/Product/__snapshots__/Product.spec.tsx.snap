// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Procduct Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-32 p-4"
  >
    <div
      class="p-4 bg-gray-100 rounded-md"
    >
      <img
        alt="Product"
        class=""
        src="/icons/product.jpeg"
      />
    </div>
    <div>
      <div
        class=" p-2"
      >
        <span
          class="text-blue-400 font-semibold"
        >
          Â£ 10
        </span>
      </div>
      <div>
        <span
          class="text-gray-900 font-semibold leading-3 text-sm"
        >
          test
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
